File: rotate-image.ts
{
    function rotate(matrix: number[][]): void {
        const result = [];

        for (let i = matrix.length - 1; i >= 0; i--) {
            for (let j = 0; j < matrix[i].length; j++) {
                if (!result[j]) {
                    result[j] = [matrix[i][j]]
                } else {
                    result[j].push(matrix[i][j])
                }

            }
        }

        for (let i = 0; i < matrix.length; i++) {
            matrix[i] = result[i];
        }

    };
    const test = [[1,2,3],[4,5,6],[7,8,9]]
    console.log(rotate(test))

    console.log(test)
}

File: rotate-image.rs
pub fn rotate(matrix: &mut Vec<Vec<i32>>) {
    matrix.reverse();
    for i in 1..matrix.len() {
        for j in 0..i {
            let (r1, r2) = matrix.split_at_mut(i);
            std::mem::swap(&mut r1[j][i], &mut r2[0][j]);
        }
    }
}
